docs/root/intro/arch_overview/http_connection_management.rst

```diff
@@ -111,6 +111,70 @@ previously attempted priorities.
 
 Envoy can be extended with custom retry plugins similar to how custom filters can be added.
 
+.. _arch_overview_internal_redirects:
+
+Internal redirects
+--------------------------
+
+Envoy supports handling 302 redirects internally, that is capturing a 302 redirect response,
+synthesizing a new request, sending it to the upstream specified by the new route match, and
+returning the redirected response as the response of the original request.
+
+There are two modes of redirect handling, "Internal redirects", for internal or private URLs,
+general redirects, which capture all 302s. They are configured separately via the
+:ref:`internal redirect action
+<envoy_api_field_route.RouteAction.internal_redirect_action>`
+and :ref:`redirect action
+<envoy_api_field_route.RouteAction.redirect_action>` fields in 
+route configuration respectively.
+
+Internal redirect handling is triggered via the x-envoy-internal-redirect header. Any response
```

Comment:
I think it might help to define somewhere in this doc a use case for an "internal redirect." This might help readers that might be confused as to why anyone would do this?